app/src/main/java/org/wikipedia/feed/onthisday/OnThisDayFragment.java

```diff
@@ -152,42 +152,25 @@ private void requestEvents(int month, int date) {
         eventsRecycler.setVisibility(View.GONE);
         errorView.setVisibility(View.GONE);
 
-        ServiceFactory.getRest(wiki).getOnThisDay(month + 1, date).enqueue(new Callback<OnThisDay>() {
-            @Override
-            public void onResponse(@NonNull Call<OnThisDay> call, @NonNull Response<OnThisDay> response) {
-                if (!isAdded()) {
-                    return;
-                }
-                onThisDay = response.body();
-                if (onThisDay == null || onThisDay.events().isEmpty()) {
-                    setErrorState(new RuntimeException("Incorrect response format."));
-                    return;
-                }
-                progressBar.setVisibility(View.GONE);
-                eventsRecycler.setVisibility(View.VISIBLE);
-                eventsRecycler.setAdapter(new RecyclerAdapter(onThisDay.events(), wiki));
-                List<OnThisDay.Event> events = onThisDay.events();
-                int beginningYear = events.get(events.size() - 1).year();
-                dayInfoTextView.setText(String.format(getString(R.string.events_count_text), Integer.toString(events.size()),
-                        DateUtil.yearToStringWithEra(beginningYear), events.get(0).year()));
-            }
-
-            @Override
-            public void onFailure(@NonNull Call<OnThisDay> call, @NonNull Throwable t) {
-                if (!isAdded()) {
-                    return;
-                }
-                setErrorState(t);
-            }
-        });
-    }
-
-    private void setErrorState(@NonNull Throwable t) {
-        L.e(t);
-        errorView.setError(t);
-        errorView.setVisibility(View.VISIBLE);
-        progressBar.setVisibility(View.GONE);
-        eventsRecycler.setVisibility(View.GONE);
+        disposables.add(ServiceFactory.getRest(wiki).getOnThisDay(month + 1, date)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(response -> {
+                    onThisDay = response;
+                    progressBar.setVisibility(View.GONE);
```

Comment:
Move `progressBar.setVisibility(View.GONE);` to `doFinally()`?

Reply:
Sure!